app/models/alchemy/essence_page.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Alchemy
+  class EssencePage < BaseRecord
+    acts_as_essence(
+      ingredient_column: :page,
+      preview_text_method: :name
+    )
+
+    belongs_to :page, class_name: 'Alchemy::Page', optional: true
+
+    def ingredient=(page)
+      case page
+      when /\d/
```

Comment:
Hm, this is too broad; it will match any string with a single integer in it. Maybe: `/^\d+$/`?